src/memphis.ts

```diff
@@ -385,8 +460,12 @@ class Producer {
             headers.headers.set('$memphis_connectionId', this.connection.connectionId);
             headers.headers.set('$memphis_producedBy', this.producerName);
             const subject = this.stationName.replace(/\./g, '#');
-            if (asyncProduce)
-                this.connection.brokerConnection.publish(`${subject}.final`, message, { headers: headers.headers, ackWait: ackWaitSec * 1000 * 1000000 });
+            let schemaData = this.connection.stationSchemaDataMap.get(subject);
```

Comment:
this function should support message of 2 types, uint8array(the old one) and a js object.
The code should identify the type of the message in case it is uint8 - continue as is, and in case it is an object to serialize it using the encode function